### PR TITLE
[feature] #2698: Fix unclear error message in `iroha_client`

### LINKED ISSUE #2689: [feature] Add parameters in `PermissionTokenDefinition`
Right now we have `PermissionTokenDefinition` which should be registered before any `PermissionToken` with corresponding defition can be granted.
But definition for now contains only *id*, when actual tokens can also have some parameters (i.e. which concrete asset can be burned). We need to put parameters in `PermissionTokenDefinition` too, because they should be fixed.

Without that user can forget to provide parameters but he would know that not on `Grant` step but when some permisison validator will check this token.

We need to insert parameters inside `PermissionTokenDefinition` and check that every granted `PermissionToken` with this definition contains the same parameters (and ideally check parameter types somehow too)

### PR DESCRIPTION
**Pull request by Arjentix**
### Description of the Change

* Reprhased two error messages

### Issue

* Closes #2689

### Benefits

* More helpful error message for the end users

### Possible Drawbacks

None

### CODE REVIEW

File: client/src/client.rs [hunk 1]
```diff
@@ -71,7 +71,8 @@ where
                     let res =
                         try_decode_all_or_just_decode!(VersionedPaginatedQueryResult, resp.body());
                     res.wrap_err(
-                        "Failed to decode the whole response body as `VersionedPaginatedQueryResult`",
+                        "Failed to decode response from Iroha. \
+                             Do you use compatible client version?",
```
Comment (appetrosyan) [inline client/src/client.rs:75]:
```suggestion
                             You are likely using a version of the client library that is incompatible with the version of the peer software?",
```

File: client/src/client.rs [hunk 2]
```diff
@@ -84,8 +85,10 @@ where
                         warn!("Can't decode query error, not all bytes were consumed");
                         res = QueryError::decode(&mut resp.body().as_ref());
                     }
-                    let err =
-                        res.wrap_err("Failed to decode the whole response body as `QueryError`")?;
+                    let err = res.wrap_err(
+                        "Failed to decode error-response from Iroha. \
+                         Do you use compatible client version?",
```
Comment (appetrosyan) [inline client/src/client.rs:90]:
```suggestion
                          You are likely using a version of the client library that is incompatible with the version of the peer software?",
```

**Review by appetrosyan — APPROVED** [review]

File: client/src/client.rs (hunk 1 above)
Comment (Arjentix) [reply client/src/client.rs:75]: @appetrosyan , may be remove `?` then?

Comment (appetrosyan) [reply client/src/client.rs:75]: yes. remove. my mistake.

**Review by appetrosyan — APPROVED** [review]

**Review by 6r1d — DISMISSED** [review]
LGTM

**Review by appetrosyan — APPROVED** [review]

**Review by 6r1d — APPROVED** [review]

**Pull request merged**